engine.py

```diff
@@ -85,10 +85,25 @@ def get_bishop_moves(self, r, c, moves):
 
 			return parameter(s):
 			None
- 		"""
+		"""
 
- 		##TODO
- 		pass
+		##TODO
+		if self.light_to_move:
+			direction = [(-1,-1), (1,1),(1,-1), (-1,1)] # possible direction
+			for d in direction:
+				for i in range(1,8):
```

Comment:
Use something that will scale to new data, using range(1,8) will only work if board size is 8 by 8, if for some reason someone changes the board size your code won't scale. Instead try;

for i in range(len(self.board)):